File: go-client-sdk/pkg/models/operations/getallcrmusers.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmUsersQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmUsersResponseBody struct {
	NextPageCursor *string       `json:"nextPageCursor,omitempty"`
	Users          []shared.User `json:"users,omitempty"`
}

type GetAllCrmUsersRequest struct {
	QueryParams GetAllCrmUsersQueryParams
}

type GetAllCrmUsersResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmUsersResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getallcrmusers.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmUsersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmUsersResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;

  @SpeakeasyMetadata({ data: "json, name=users", elemType: shared.User })
  users?: shared.User[];
}


export class GetAllCrmUsersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmUsersQueryParams;
}


export class GetAllCrmUsersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmUsersResponseBody;
}
